### PR TITLE
Add 'Fill background' and 'Draw frame' functionality

### PR DESCRIPTION
**Pull request by ososco**
One-click button to fill background according to picked color, and another button to draw instant frames according to picked color and radius.

image

### CODE REVIEW

File: packages/react-app/src/InkCanvas.js [hunk 1]
```diff
@@ -256,7 +256,45 @@ const onFinishFailed = errorInfo => {
...
+      { x: 0, y: 0 },
+      { x: width, y: height }
+    ]
+  });
+
+  let saved = drawingCanvas.current.getSaveData();
+  drawingCanvas.current.loadSaveData(saved, true);
+};
 
+const drawFrame = (color, radius) => {
```
Comment (azf20) [inline packages/react-app/src/InkCanvas.js:276]: this is great

File: packages/react-app/src/InkCanvas.js [hunk 2]
```diff
@@ -256,7 +256,45 @@ const onFinishFailed = errorInfo => {
   console.log('Failed:', errorInfo);
 };
 
+const fillBackground = (color) => {
```
Comment (azf20) [inline packages/react-app/src/InkCanvas.js:259]: Nice. I wonder if there is a way to inject this so it's a true background (currently it's definitely a fill, but it kind of fills in everything)

File: packages/react-app/src/InkCanvas.js [hunk 3]
```diff
@@ -353,6 +391,18 @@ if (props.mode === "edit") {
           />
         </Col>
     </Row>
+    <Row style={{ width: "90vmin", margin: "0 auto", marginTop:"4vh", justifyContent:'center'}}>
+        <Col span={4}>
```
Comment (azf20) [inline packages/react-app/src/InkCanvas.js:395]: I got a bit of overlap between these buttons, I think you can remove the Col wrapper (it's not doing too much)
image

File: packages/react-app/src/InkCanvas.js [hunk 4]
```diff
@@ -353,6 +391,18 @@ if (props.mode === "edit") {
...
+          <Button
+          onClick={() => fillBackground(color)}
+          ><BgColorsOutlined />Background</Button>
```
Comment (azf20) [inline packages/react-app/src/InkCanvas.js:398]: Should we maybe call this "Fill" rather than background?

**Review by azf20 — COMMENTED** [review]
Really nice! Simple elegant implementation. Couple of quick comments

File: packages/react-app/src/InkCanvas.js (hunk 2 above)
Comment (dhalsim) [reply packages/react-app/src/InkCanvas.js:259]: maybe that could be possible if we forked react-canvas-draw and changed every code from

```
ctx.clearRect(0, 0, ctx.canvas.width, ctx.canvas.height);
```

to fill the color we passed (there is no option to pass either)

```
ctx.fillStyle = FILL_COLOR;
ctx.fillRect(0, 0, ctx.canvas.width, ctx.canvas.height);
```

I think Osama's approach is better, very nice workaround.

Comment (azf20) [reply packages/react-app/src/InkCanvas.js:259]: I guess I meant more to show up behind any drawing that has already been done. But there is a lot of complexity there I think

Comment (dhalsim) [reply packages/react-app/src/InkCanvas.js:259]: Actually there can be an easy way to make this always stay on the background. Maybe I can check it this evening if that works. basically if we manage to always draw background before everything else that would solve our problem

**Pull request merged**